packages/engine-test-utils/src/__tests__/engine-server/markdown/blockAnchors.spec.ts

```diff
@@ -0,0 +1,96 @@
+import {
+  DendronASTData,
+  DendronASTDest,
+  DEngineClient,
+  MDUtilsV4,
+  UnistNode,
+  BlockAnchor,
+  blockAnchors,
+  BlockAnchorOpts,
+  DendronASTNode,
+} from "@dendronhq/engine-server";
+import _ from "lodash";
+
+function proc(
+  engine: DEngineClient,
+  dendron: DendronASTData,
+  opts?: BlockAnchorOpts
+) {
+  return MDUtilsV4.proc({ engine })
+    .data("dendron", dendron)
+    .use(blockAnchors, opts);
+}
+
+function genDendronData(opts?: Partial<DendronASTData>): DendronASTData {
+  return { ...opts } as any;
+}
+
+/** Gets the descendent (child, or child of child...) node of a given node.
+ *
+ * @param node The root node to start descending from.
+ * @param indices Left-to-right indexes for children, e.g. first index is for the root, second is for the child of the root...
+ * @returns Requested child. Note that this function has no way of checking types, so the child you get might not be of the right type.
+ */
+function getDescendentNode<Child extends DendronASTNode>(
```

Comment:
maybe rename to `getLeafNode`? that's a more widely used term for this

Reply:
I'll argue that "descendant" is the right term here. The node that gets selected won't necessarily be a leaf node, it may have children of it's own. It's a relatively widely used term as well, here are a few examples of the term being used:

https://developer.mozilla.org/en-US/docs/Web/CSS/Descendant_combinator

https://api.jquery.com/descendant-selector/

I can change it of course if you don't agree that it's appropriate.

Reply:
your right, misread the function. this is good, thanks for the examples :)